### PR TITLE
CGE-88 Update personal information fixes

### PR DESCRIPTION
**Pull request by lubchik-14**
Fixing according to testers notes
- It is necessary to remove the ID line from the request body (Xenia Ostolosh)
- Users should not be able to change their last online time, so the "lastOnlineTime" field should not be editable (Dmitriy)

### CODE REVIEW

File: src/main/java/com/hillel/items_exchange/dto/UserDto.java [hunk 1]
```diff
@@ -38,6 +36,6 @@
     private String avatarImage;
     @NotNull(message = "{invalid.not-null}")
     @PastOrPresent(message = "{invalid.past-or-present.date}")
-    @JsonFormat(pattern = "yyyy-MM-dd")
-    private LocalDate lastOnlineTime;
+    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
+    private LocalDateTime lastOnlineTime;
 }
```
Comment (rpkyrych) [inline src/main/java/com/hillel/items_exchange/dto/UserDto.java:41]: @lubchik-14, could you add list of the phones and children to UserDTO? @Wolshebnik is started working on the personal area and he needs that data into one request.

File: src/test/java/com/hillel/items_exchange/controller/UserControllerTest.java [hunk 2]
```diff
@@ -111,20 +111,20 @@ void updateUserInfo_shouldReturn403WhenUsernameIsChanged() throws Exception {
...
     @Transactional
     @DataSet("database_init.yml")
-    void updateUserInfo_shouldBeThrownAccessDeniedExceptionWhenNotUserId() throws Exception {
+    void updateUserInfo_shouldReturn403WhenLastOnlineTimeIsChanged() throws Exception {
         MvcResult result = getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithNotUserId(), status().isConflict())
+                createUserDtoForUpdatingWithChangedLastOnlineTimeWithoutChildrenOrPhones(), status().isForbidden())
                 .andReturn();
-        assertTrue(result.getResponse().getContentAsString().contains("You are not allowed to perform this action"));
+        assertTrue(result.getResponse().getContentAsString().contains("You are unable to change your: LastOnlineTime"));
```
Comment (rpkyrych) [inline src/test/java/com/hillel/items_exchange/controller/UserControllerTest.java:118]: You could export such message into messages.properties and get from it. So you won't need to update it every time it was updated there.

File: src/test/java/com/hillel/items_exchange/util/UserDtoCreatingUtil.java [hunk 3]
```diff
@@ -1,55 +1,66 @@
...
     public static final String NEW_INVALID_SHORT_NAME = "n";
     public static final String NEW_INVALID_TWO_WORDS_NAME = "new Name";
 
-    public static UserDto createUserDtoForUpdatingWithChangedEmailAndFirstNameApostrAndLastNameMinus() {
-        return getBuild(1L, "admin", NEW_EMAIL, Boolean.FALSE, NEW_VALID_NAME_WITH_APOSTROPHE,
-                NEW_VALID_NAME_WITH_HYPHEN_MINUS, "empty", LocalDate.of(2019, 1, 1));
+    public static UserDto createUserDtoForUpdatingWithChangedEmailAndFNameApAndLNameMinusWithoutChildrenOrPhones() {
+        return getBuild("admin", NEW_EMAIL, Boolean.FALSE, NEW_VALID_NAME_WITH_APOSTROPHE,
+                NEW_VALID_NAME_WITH_HYPHEN_MINUS, "empty", LocalDateTime.of(2019, 1, 1, 0, 0, 1),
+                Collections.emptySet(), Collections.emptySet());
```
Comment (rpkyrych) [inline src/test/java/com/hillel/items_exchange/util/UserDtoCreatingUtil.java:24]: For now let it be with empty collections but soon we need replace them with real data.

**Review by rpkyrych — APPROVED** [review]
Thanks for updating it.

**Review by Jack11M — APPROVED** [review]

**Pull request merged**